app/controllers/incomeSources/add/CheckBusinessDetailsController.scala

```diff
@@ -180,4 +173,50 @@ class CheckBusinessDetailsController @Inject()(val checkBusinessDetails: CheckBu
         }
   }
 
+  private def getBusinessDetailsFromSession(implicit user: MtdItUser[_], ec: ExecutionContext): Future[CheckBusinessDetailsViewModel] = {
+    val userActiveBusinesses: List[BusinessDetailsModel] = user.incomeSources.businesses.filterNot(_.isCeased)
+    val skipAccountingMethod: Boolean = userActiveBusinesses.isEmpty
+
+    val result = for {
+      businessName <- sessionService.get(SessionKeys.businessName)
+      businessStartDate <- sessionService.get(SessionKeys.businessStartDate)
+      businessTrade <- sessionService.get(SessionKeys.businessTrade)
+      businessAddressLine1 <- sessionService.get(SessionKeys.addBusinessAddressLine1)
+      businessAccountingMethod <- sessionService.get(SessionKeys.addIncomeSourcesAccountingMethod)
+      accountingPeriodEndDate <- sessionService.get(SessionKeys.addBusinessAccountingPeriodEndDate)
+      businessAddressLine2 <- sessionService.get(SessionKeys.addBusinessAddressLine2)
+      businessAddressLine3 <- sessionService.get(SessionKeys.addBusinessAddressLine3)
+      businessAddressLine4 <- sessionService.get(SessionKeys.addBusinessAddressLine4)
+      businessPostalCode <- sessionService.get(SessionKeys.addBusinessPostalCode)
+      businessCountryCode <- sessionService.get(SessionKeys.addBusinessCountryCode)
+      incomeSourcesAccountingMethod <- sessionService.get(SessionKeys.addIncomeSourcesAccountingMethod)
+    } yield (businessName, businessStartDate, businessTrade, businessAddressLine1,
+      businessAccountingMethod, accountingPeriodEndDate, businessAddressLine2,
+      businessAddressLine3, businessAddressLine4, businessPostalCode, businessCountryCode,
+      incomeSourcesAccountingMethod)
+
+    result.flatMap {
+      case (Right(businessName), Right(businessStartDate), Right(Some(businessTrade)),
+      Right(Some(businessAddressLine1)), Right(Some(businessAccountingMethod)), Right(accountingPeriodEndDate),
+      Right(businessAddressLine2), Right(businessAddressLine3), Right(businessAddressLine4), Right(businessPostalCode),
+      Right(businessCountryCode), Right(incomeSourcesAccountingMethod)) =>
+        Future.successful(CheckBusinessDetailsViewModel(
+          businessName = businessName,
+          businessStartDate = businessStartDate.map(LocalDate.parse(_)),
+          accountingPeriodEndDate = accountingPeriodEndDate.map(LocalDate.parse(_)).get,
```

Comment:
if accountingPeriodEndDate is mandatory field, we need to handle parsing in a right way
Please don't call .get directly over Option

Reply:
Handled the parsing for mandatory field